crates/chain/src/tx_graph.rs

```diff
@@ -1161,6 +1163,50 @@ impl<A: Anchor> TxGraph<A> {
         self.try_balance(chain, chain_tip, outpoints, trust_predicate)
             .expect("oracle is infallible")
     }
+
+    /// Returns an iterator over unconfirmed transactions and their associated script pubkeys,
+    /// filtered within the specified `range`.
+    ///
+    /// This function scans the transaction graph for unconfirmed transactions relevant to the
+    /// provided [`SpkTxOutIndex`], determining which transactions should be considered based on
+    /// indexed outputs.
+    pub fn iter_spks_with_expected_txids<'a, C, I>(
+        &'a self,
+        chain: &'a C,
+        indexer: &'a SpkTxOutIndex<I>,
+        range: impl RangeBounds<I> + 'a,
+    ) -> impl Iterator<Item = (Txid, ScriptBuf)> + 'a
+    where
+        C: ChainOracle<Error = core::convert::Infallible>,
+        I: fmt::Debug + Clone + Ord,
+    {
+        let chain_tip = chain.get_chain_tip().unwrap();
+
+        self.list_canonical_txs(chain, chain_tip)
+            .filter(|c| !c.chain_position.is_confirmed() && indexer.is_tx_relevant(&c.tx_node))
```

Comment:
Any confirmed tx that is missing from the Electrum's spk history should also be evicted. Since this means that there is a reorg and the tx is not re-introduced to the mempool (evicted).